File: examples/typescript/testing-patterns/basic-tests/test-error-handling.ts
import {
  LocalDurableTestRunner,
  OperationStatus,
  OperationType,
} from "@aws/durable-execution-sdk-js-testing";
import { ExecutionStatus } from "@aws-sdk/client-lambda";
import { handler } from "./add-numbers";

let runner: LocalDurableTestRunner;

beforeAll(async () => {
  await LocalDurableTestRunner.setupTestEnvironment({ skipTime: true });
});

afterAll(async () => {
  await LocalDurableTestRunner.teardownTestEnvironment();
});

beforeEach(() => {
  runner = new LocalDurableTestRunner({ handlerFunction: handler });
});

it("should inspect step result by name", async () => {
  const stepOp = runner.getOperation("add_numbers");

  const result = await runner.run();

  expect(result.getStatus()).toBe(ExecutionStatus.SUCCEEDED);

  const details = stepOp.getStepDetails();
  expect(details?.result).toBeDefined();
  expect(stepOp.getStatus()).toBe(OperationStatus.SUCCEEDED);
});

File: examples/java/testing-patterns/basic-tests/test-error-handling.java
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;
import software.amazon.awssdk.services.lambda.model.OperationStatus;
import software.amazon.lambda.durable.model.ExecutionStatus;
import software.amazon.lambda.durable.testing.LocalDurableTestRunner;

class InspectStepResultsTest {
    @Test
    void testInspectStepResult() {
        var runner = LocalDurableTestRunner.create(Object.class, new AddNumbersExample());

        var result = runner.runUntilComplete(null);

        assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());

        var stepOp = runner.getOperation("add_numbers");
        assertNotNull(stepOp);
        assertEquals(OperationStatus.SUCCEEDED, stepOp.getStatus());
        assertEquals(8, stepOp.getStepResult(Integer.class));
    }
}
